internal/powerplatform/services/data_record/api_data_record.go

```diff
@@ -0,0 +1,432 @@
+	}
+
+	return &env, nil
+}
+
+func (client *DataRecordClient) GetDataRecord(ctx context.Context, recordId string, environmentId string, tableName string) (map[string]interface{}, error) {
+	environmentUrl, err := client.GetEnvironmentUrlById(ctx, environmentId)
+	if err != nil {
+		return nil, err
+	}
+
+	entityDefinition := getEntityDefinition(ctx, client, environmentUrl, tableName)
+
+	e, _ := url.Parse(environmentUrl)
+	apiUrl := &url.URL{
+		Scheme: e.Scheme,
+		Host:   e.Host,
+		Path:   fmt.Sprintf("/api/data/%s/%s(%s)", constants.DATAVERSE_API_VERSION, entityDefinition.LogicalCollectionName, recordId),
+	}
+
+	result := make(map[string]interface{}, 0)
+
+	_, err = client.Api.Execute(ctx, "GET", apiUrl.String(), nil, nil, []int{http.StatusOK}, &result)
+	if err != nil {
+		if strings.ContainsAny(err.Error(), "404") {
+			return nil, powerplatform_helpers.WrapIntoProviderError(err, powerplatform_helpers.ERROR_OBJECT_NOT_FOUND, fmt.Sprintf("Data Record '%s' not found", recordId))
+		}
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (client *DataRecordClient) GetRelationData(ctx context.Context, recordId string, environmentId string, tableName string, relationName string) ([]interface{}, error) {
+	environmentUrl, err := client.GetEnvironmentUrlById(ctx, environmentId)
+	if err != nil {
+		return nil, err
+	}
+
+	entityDefinition := getEntityDefinition(ctx, client, environmentUrl, tableName)
+
+	e, _ := url.Parse(environmentUrl)
+	apiUrl := &url.URL{
+		Scheme:   e.Scheme,
+		Host:     e.Host,
+		Path:     fmt.Sprintf("/api/data/%s/%s(%s)/%s", constants.DATAVERSE_API_VERSION, entityDefinition.LogicalCollectionName, recordId, relationName),
+		RawQuery: "$select=createdon",
+	}
+
+	result := make(map[string]interface{}, 0)
+
+	_, err = client.Api.Execute(ctx, "GET", apiUrl.String(), nil, nil, []int{http.StatusOK}, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	field, ok := result["value"]
+	if !ok {
+		return nil, fmt.Errorf("value field not found in result")
+	}
+
+	value, ok := field.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("value field is not of type []interface{}")
+	}
+
+	return value, nil
+}
+
+func (client *DataRecordClient) GetEntityRelationDefinitionInfo(ctx context.Context, environmentId string, entityLogicalName string, relationLogicalName string) (tableName string) {
+	environmentUrl, err := client.GetEnvironmentUrlById(ctx, environmentId)
+	if err != nil {
+		return ""
+	}
+
+	apiUrl := fmt.Sprintf("%s/api/data/%s/EntityDefinitions(LogicalName='%s')?$expand=OneToManyRelationships,ManyToManyRelationships,ManyToOneRelationships", environmentUrl, constants.DATAVERSE_API_VERSION, entityLogicalName)
+
+	response, err := client.Api.Execute(ctx, "GET", apiUrl, nil, nil, []int{http.StatusOK}, nil)
+	if err != nil {
+		return ""
+	}
+
+	var mapResponse map[string]interface{}
+	json.Unmarshal(response.BodyAsBytes, &mapResponse)
+
+	oneToMany, _ := mapResponse["OneToManyRelationships"].([]interface{})
```

Comment:
I would validate if "OneToManyRelationships" and other string exist as keys in the map. What happens in a table does not have this type of relationship at all in metdata?

Reply:
done 👍